runtime/util/srphashtable.c

```diff
@@ -621,7 +639,9 @@ srpHashTableForEachDo(J9SRPHashTable *srptable, J9SRPHashTableDoFn doFn, void *o
 
 	Assert_srphashtable_true(NULL != nodes);
 
+#ifdef SRPHASHTABLE_DEBUG
```

Comment:
I am just curious what the compiler error was for this change.  Was it complaining about calling an empty macro if the debug macro was not defined?

Reply:
srpHashTable_printf ought to have been a function-style macro.

If all the uses of `srpHashTable_printf` will be guarded by `#ifdef SRPHASHTABLE_DEBUG`, perhaps `srpHashTable_printf` should be removed?

Reply:
Here's an example of the errors:
```
        srpHashTable_printf(PORTLIB, "srpHashTableNew <%s>: tableSize=%d, srpTable=%p\n", tableName, tableSize, srpHashTable);
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
srphashtable.c:161:84: error: expression result unused [-Werror,-Wunused-value]
        srpHashTable_printf(PORTLIB, "srpHashTableNew <%s>: tableSize=%d, srpTable=%p\n", tableName, tableSize, srpHashTable);
                                                                                          ^~~~~~~~~
srphashtable.c:161:95: error: expression result unused [-Werror,-Wunused-value]
        srpHashTable_printf(PORTLIB, "srpHashTableNew <%s>: tableSize=%d, srpTable=%p\n", tableName, tableSize, srpHashTable);
                                                                                                     ^~~~~~~~~
srphashtable.c:161:106: error: expression result unused [-Werror,-Wunused-value]
        srpHashTable_printf(PORTLIB, "srpHashTableNew <%s>: tableSize=%d, srpTable=%p\n", tableName, tableSize, srpHashTable);
                                                                                                                ^~~~~~~~~~~~
```

Reply:
Why would we remove the debugging code?  Having this code here in all the relevant places makes it easier to debug the srphashtable in the future.

Reply:
I meant we could remove the macro, not the debug code. The macro is only used if debugging is enabled; we could use j9tty_printf directly.

Alternatively we could change the macro definition so none of the new ifdefs are required.